src/routes/auth/index.ts

```diff
@@ -102,24 +102,26 @@ authRoutes.post('/sign-in', async (context: Context<EnvironmentBindings>) => {
 		throw error;
 	}
 
-	const { id: profileId, password: storedPassword } = await getProfileIdPassword(context.env.database, parsedBody.email);
-	if (!profileId || !storedPassword) {
-		return context.json<StatusResponse>({ message: 'Invalid email' }, StatusCodes.UNAUTHORIZED);
-	}
+	const results = await getLoginInfo(context.env.database, parsedBody.email);
 
-	const isProfileValid = await checkProfile(context.env.database, profileId);
-	if (!isProfileValid) {
-		return context.json<StatusResponse>({ message: 'Invalid profile id or Account not activated' }, StatusCodes.UNAUTHORIZED);
+	const genericSignInError = 'Either your email/password combination is invalid, or your account is not active';
```

Comment:
I'm not a huge fan of these error messages that encompasses several possible error scenario but in this case I cant think of a way around.